### DISCUSSION TITLE
identify takes long for (physically) large pdfs

### DISCUSSION THREAD

**Discussion by leifg** [ImageMagick/ImageMagick#2993] [Help] [answered]
Hello there,

I'm using the `identify` command of imagemagick to identify the physical page size of a PDF document.

```
identify -verbose document.pdf | grep 'Print size'
```

It seems like when the print size is too large identify takes a long time (I have one example that takes an hour).

This has nothing to do with the file size, here is how to reproduce a PDF that takes 5 min to identify (on my machine, current 2.3 GHz 8-Core Intel Core i9 MacBook Pro 16" with 64GB of RAM):

- Take any reasonably sized picture you can find (I took a 881x1322 pixels jpg which is 161kb file size)
- convert this into a PDF with a density of 1 DPI via convert:
```shell
convert regular_size_image.jpg -density 1 comically_large_document.pdf
``` 
-  This will create a 881 x 1322 inch PDF document

The file size of this document is appropriately the same file size as the input image and can be opened with any PDF viewer and the physical dimensions are shown.

running any `identify` command on the resulting PDF takes ~5 minutes. I added the document for I was using for reproducibility:

[comically_large_document.pdf](https://github.com/ImageMagick/ImageMagick/files/5697769/comically_large_document.pdf)

So I'm wondering if this is a problem within imagemagick (my version string is `ImageMagick 7.0.10-48 Q16 x86_64 2020-12-10 https://imagemagick.org`).

Alternatively is there anything I can do to just fetch the print size that doesn't take this long?

**Comment by snibgo** [comment]
When IM reads PDF fles, it tells Ghostscript to rasterize the image at whatever density you give, default 72 DPI. So if your image is 881x1322 inches, this is 6 giga-pixels, probably too many for your memory, so it takes a long time.

One solution is to rasterize at a lower DPI, eg 10, and adjust the result accordingly.

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by leifg** [comment]
Just tested with `imagemagick 7.0.10-53` and the identifying is significantly faster now. Thank you!

The commit that fixes it is [here](https://github.com/ImageMagick/ImageMagick/commit/49d662c50dfeadbc09ec60cf28f7044e8780a292)
